docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+### Image decryption
+
+After verifying the container image signature, will start the actual image pulling. When pulling the image layer by layer, if encounter an encrypted image layer, image-rs calls the container image cryptography operation library provided by ocicrypt-rs to decrypt the encrypted image layer as follows:
+
+1. Read the layer's two annotations fields in the image manifest. The `org.opencontainers.image.enc.keys.provider.attestation` field is the annotation packet mentioned above, which contains `PrivateLayerBlockCipherOptions` encrypted by the user's private key. The symmetric key used to encrypt this layer is recorded in this `PrivateLayerBlockCipherOptions`.
+2. ocicrypt-rs calls the keyprovider grpc service provided by the attestation-agent and sends the annotation packet to the attestation-agent.
+3. The attestation-agent performs HW-TEE remote attestation with the relying KBS deployed by the remote user. After passing the attestation, it interacts with KBS through the encrypted channel, and decrypts the `PrivateLayerBlockCipherOptions` using the user's public key in the KBS.
+4. ocicrypt-rs obtains the decrypted `PrivateLayerBlockCipherOptions` from the attestation-agent, reads the symmetric key used to decrypt the container image, and completes the decryption of the container image layer.
+
+## Attestation Agent
+
+The attestation-agent is an indispensable core component in the confidential containers architecture. It undertakes the trust distribution function of the confidential container. In the process of signature verification and decryption of the protected image, the attestation-agent serves as the source of the user's confidential information and the decryptor of the `PrivateLayerBlockCipherOptions`. As mentioned above, it provides two gRPC services for image-rs and ocicrypt-rs to support the above two functions.
+
+### GetResource service
+
+The GetResource grpc service interface provided by the attestation agent is used to obtain various confidential resources from the relying party (KBS). In the process of protected image deployment, it is usually used to obtain policiy.json and sigstore config file. 
+
+Attestation-agent performs HW-TEE attestation to the KBS, establish an encrypted channel after negotiating the key, and download the required confidential resources through the encrypted channel. The GetResource grpc service interface protobuf of the attachment agent is defined as follows:
+
+```protobuf
+message getResourceRequest {
+    string KbcName = 1;
+    string KbsUri = 2;
+    string ResourceDescription = 3;
+}
+
+message getResourceResponse {
+    bytes resource = 1;
+}
+
+service GetResource {
+    rpc GetResource(getResourceRequest) returns (getResourceResponse) {};
+}
+```
+
+ResourceDescription is a JSON string used to describe resources, and its format is defined as follows:
+
+```json
+{
+    "name":"example_resource",
+    "informations":{
```

Comment:
When you transfer a resource, the source actually was something like a file (e.g. `policy.json`). Does the `ResourceDescription` give you a URL to that file, or the data itself embedded in the JSON?